### PR TITLE
Add redux tests

### PR DESCRIPTION
**Pull request by sabicd**
- Remove extra lines from `app` store

- add tests to the project for redux

- add comments to easier navigation in the test files

### CODE REVIEW

File: src/__tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  it('toggleMobileDrawer', () => {
+    const expectedAction = {
+      type: app.TOGGLE_MOBILE_DRAWER,
+    };
+    expect(app.toggleMobileDrawer()).toEqual(expectedAction);
+  });
+  it('setAppWaiting', () => {
+    const expectedAction = {
+      type: app.SET_APP_WAITING,
+    };
```
Comment (knightburton) [inline src/__tests__/app.test.js]: The payload is missing from this. That could be true or false. I mean the undefines is still something that can occure, however we need to test the other possibilities as well.

File: src/__tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      payload: 1
+    };
+    expect(app.removeAlert(1)).toEqual(expectedAction);
+  });
+});
+/**
+ * SELECTORS
+ */
+describe('selectors', () => {
+  it('getIsMobileDrawerOpened', () => {
```
Comment (knightburton) [inline src/__tests__/app.test.js]: I suggest to define a local state over here inside the selector describer, and you do not have to declare a separated app store for each value. And use that as a parameter of a selector.
```javascript
const state = {
  app: {
   ...
  },
};
```

**Review by knightburton — CHANGES_REQUESTED** [review]
In general if you put a store test file into the __test__ folder then put them inside a store folder as well. In the future we can identify them more easliy if we add more tests or ui tests as well.

File: src/__tests__/app.test.js (hunk 2 above)
Comment (sabicd) [reply src/__tests__/app.test.js]: I've wanted to differentiate the state for each test, but I also agree with this. Modifying it.

File: src/__tests__/app.test.js (hunk 1 above)
Comment (sabicd) [reply src/__tests__/app.test.js]: > In general if you put a store test file into the **test** folder then put them inside a store folder as well. In the future we can identify them more easliy if we add more tests or ui tests as well.

sure, moving them under a `store` directory

**Review by knightburton — APPROVED** [review]

**Pull request merged**